lib/fs.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const { promisify } = require('util');
+
+const copyFile = promisify(fs.copyFile);
+const symlink = promisify(fs.symlink);
+const readlink = promisify(fs.readlink);
+const lstat = promisify(fs.lstat);
+const readdir = promisify(fs.readdir);
+const mkdir = promisify(fs.mkdir);
+const rmdir = promisify(fs.rmdir);
+const unlink = promisify(fs.unlink);
+
+const rmDir = (current, callback) => {
+  const removeDir = current =>
+    new Promise((resolve, reject) =>
+      readdir(current)
+        .then(items =>
+          Promise.all(
+            items.map(item =>
+              lstat(path.resolve(current, item)).then(stat => {
+                if (stat.isDirectory())
+                  return removeDir(path.join(current, item));
+                else return unlink(path.join(current, item));
+              })
+            )
+          )
+        )
+        .then(() => rmdir(current))
+        .catch(err => reject(err))
+        .then(resolve)
+    );
+  removeDir(current)
+    .then(() => callback(null))
+    .catch(err => callback(err));
+};
+
+const copyDir = (source, destination, options, callback) => {
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+  const { depth, filter, errorLog, dereference } = options;
+  const { Transform, readOptions, writeOptions } = options;
+
+  const error = (err, resolve, reject) => {
+    if (err.code === 'EEXIST') resolve();
+    else if (!errorLog) reject(err);
+    else {
+      errorLog.write(JSON.stringify(err));
+      resolve();
+    }
+  };
+
+  const cpFile = (current, target) => {
```

Comment:
This should be moved outside the function or inlined as you use it only once. Also, in this case, it is better to pass variables you need as variables then rely on context.